### ISSUE TITLE
[Bug]: IndexError: list index out of range 

### ISSUE THREAD

**Issue by shiyu22** [towhee-io/towhee#2445]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When I run the following code, will show `IndexError: list index out of range `:
```python
from towhee.dc2 import pipe, ops, DataCollection

obj_filter_embedding = (
    pipe.input('url')
        .map('url', 'img', ops.image_decode.cv2_rgb())
        .map('img', 'obj_res', ops.object_detection.yolo())
        .filter(('img', 'obj_res'), ('img', 'obj_res'), 'obj_res', lambda x: len(x) > 0)
        .output('url', 'img', 'obj_res')
    )

res = obj_filter_embedding('https://towhee.io/object-detection/yolo/raw/branch/main/logo.png')
DataCollection(res).show()
```

### Expected Behavior

Return none instead of error.

### Environment

```markdown
- Towhee version(e.g. v0.1.3 or 8b23a93):
- OS(Ubuntu or CentOS):
- CPU/Memory:
- GPU:
- Others:
```

**Comment by junjiejiangjjj** [comment]
`DataCollection` can't handle an empty `dataqueue` correctly.  @Chiiizzzy

**Comment by binbinlv** [comment]
Branch 1.0 has been created, so if any fix in towhee is needed, please fix both in main and branch 1.0, thanks.
@junjiejiangjjj @Chiiizzzy
